### PR TITLE
md parser regression: setext headings no longer work #155

### PR DESCRIPTION
**Pull request by rofe**
As discussed in #155, I removed the line im parse-markdown.js that caused setext headings to be ignored. Also added a test to show that both setext and # headings are processed correctly.

### CODE REVIEW

**Comment by rofe** [comment]
Fixes #155

File: src/html/parse-markdown.js [hunk 1]
```diff
@@ -39,9 +39,6 @@ const thbreak = {
 function parse({ content: { body = '' } = {} }, { logger }) {
   logger.debug(`Parsing markdown from request body starting with ${body.split('\n')[0]}`);
 
-  // disable setext headings (=== and ---)
-  remark.Parser.prototype.blockTokenizers.setextHeading = () => {};
-
   const preprocessor = unified()
     .use(remark, { setext: false })
```
Comment (trieloff) [inline src/html/parse-markdown.js]: Does this `{ setext: false }` flag do anything?

Comment (rofe) [reply src/html/parse-markdown.js]: I was wondering the same thing actually. Our tests pass with and without this option, but I wasn't entirely certein and wanted to keep my changes as surgical as possible. `remark` [docs](https://github.com/remarkjs/remark/tree/master/packages/remark-stringify#optionssetext) suggest it does enable or disable setext headings, though...

Comment (rofe) [reply src/html/parse-markdown.js]: That appears to be a `remark-stringify` option :) I think we can remove it.

**Comment by rofe** [comment]
> Looks good. It would be great to test setext headlines in conjunction with YAML front matter and sections, too.

I tried replacing all level 1 and 2 atx headings in `frontmatter.md`(https://github.com/adobe/helix-pipeline/blob/master/test/fixtures/frontmatter.md) with setext format. Parsing works for all but [one](https://github.com/adobe/helix-pipeline/blame/master/test/fixtures/frontmatter.md#L144) which gets rendered as a `paragraph` followed by a `thematicBreak`...

**Comment by rofe** [comment]
As discussed in [#155](https://github.com/adobe/helix-pipeline/issues/155#issuecomment-464837545) let's handle `frontmatter` in a separate upstream issue.

**Pull request merged**